### ISSUE TITLE
GetAuditReadyDiff does not ignore private/protected complex properties/fields

### ISSUE THREAD

**Issue by IceAndSnow** [rapidcore/rapidcore#136]
`GetAuditReadyDiff` should not report changes on private or protected fields/properties. This is however done for complex fields/properties. An example test which fails is below.

```
[Fact]
public void GetAuditReadyDiff_doesNotIncludePrivateComplexProperty()
{
    var oldState = new ThingWithPrivateProperty
    {
        NotPrivateProperty = "Old string"
    }.SetPrivate(new UnimportantThing {SomeString = "Old unimportant string"});
            
    var newState = new ThingWithPrivateProperty
    {
        NotPrivateProperty = "New string"
    }.SetPrivate(new UnimportantThing {SomeString = "New unimportant string"});
            
    var actual = auditDiffer.GetAuditReadyDiff(oldState, newState);

    Assert.Single(actual.Changes);

    Assert.Equal("ThisShouldBeInTheLog", actual.Changes[0].Breadcrumb);
    Assert.Equal("Old string", actual.Changes[0].OldValue);
    Assert.Equal("New string", actual.Changes[0].NewValue);
}

public class ThingWithPrivateProperty
{
    public string NotPrivateProperty { get; set; }
    private UnimportantThing PrivateProperty { get; set; }

    public ThingWithPrivateProperty SetPrivate(UnimportantThing p)
    {
        PrivateProperty = p;
        return this;
    }
}
```

A solution to this is to add a private/protected check in the `ShouldIgnoreMember` method in `AuditDiffer` to replace the existing check. After #129 is solved.

**Comment by pingvinen** [comment]
> should not report changes on private or protected fields/properties

I do not agree with that statement. You could have state in private/protected members that is valuable in an audit log. If you do not want them in your particular log, you can decorate them with `[Audit(Include=false)]`.

A possible middle ground would be to add a system of some kind for providing configuration, wherein some of those config values could control whether or not to include private and protected members.

**Comment by IceAndSnow** [comment]
I see your point, however, in that case I believe there is a problem with the documentation, unless I am misreading it.
This is a quote from the `AuditDiffer.md`:

> The `AuditDiffer` is an object differ, specifically built for generating diffs that can go into an audit log.
>
> In list form it...
> 
> - will find changes made to `public` fields and properties
> - allows you to exclude certain fields or properties using the `AuditAttribute`
> - allows you to "mask" certain fields or properties using the `AuditAttribute` to set an [IAuditValueMasker](../ValueMasking)

**Comment by IceAndSnow** [comment]
Also, what I actually mean by this issue is that the behavior differs for complex types and primitive types.
Right now, private primitive properties are not included in the diff (e.g. `private string NotIncluded { get; set; }`).
This is not the case for complex types, as they always seem to be included (unless explicitly ignored) (e.g. `private UnimportantThing ThisIsActuallyIncluded { get; set; }`)

I would expect the behavior to at least be the same no matter if the type is primitive or complex

**Comment by pingvinen** [comment]
Ok, we will have to look into that!

As I recall, the trick is to not reflect into the primitive types, so it might be some protection against that, which is too broad.

I suggest that we:

1. update the documentation, so it is explicit that we also find changes in private and protected members
2. look into the behavior and ensure that it is consistent (specifically, I think we should add a test that has all the various members and specifically looks for consistent behavior)
